### ISSUE TITLE
[GBA] Save file no longer works and all my save states are gone.

### ISSUE THREAD

**Issue by ghostrule** [mgba-emu/mgba#1112]
Version: 0.6.1
OS: Windows 7, 64-bit
Other stuff... Guideline said this might not be necessary, I'm using AMD Radeon something something.

I have been using this emulator to play MegaMan Battle Network, the very first one. It had it's own save file, the .sav file, and it also had a number of save states. I've been playing it and streaming it for a bit without any problems, until I set it down for a few weeks and today when I tried to play it again, I found that all of my save states were gone, and when I attempted to load the save, the screen turned completely white and nothing else happened until I closed the emulator and opened it back up. 

Asides from moving another game into the same folder (it also was a .gba and .sav file, but I earlier removed them- and yes, I'm absolutely positive that they weren't my Battle Network files :P), nothing had changed to this folder or the files in it, so I'm completely clueless as to what the problem could be. My friends suggest that it got corrupted somehow since the screen goes white, but I really don't know. (I'm also doubting it, but it might just be my shred of hope that I don't have to completely restart now.)

I looked around a bit but I couldn't find anything relevant to my problem, so I decided to post it here. If anyone knows what happened, or if I'm completely screwed on my save file or something, I'd really appreciate a response. Thank's in advance.

**Comment by Dimensional** [comment]
Since you haven't used the emulator for a long time, it sounds like the problem isn't the emulator but something related to the OS, which means writing the issue here wouldn't mean a thing. Maybe you accidentally were cleaning up space on the drive with some automated program and it decided to erase those? Or you had an abrupt loss of power, which could corrupt random portions of your storage. In fact, depending on what type of storage you're using, wear and tear could also have happened. The possibilities are numerous, but again it's not an mgba issue as far as you're explanation states.

**Comment by endrift** [comment]
Can you upload the savestates?

**Comment by ghostrule** [comment]
Well in that case, then, oops! Writing it here wouldn't mean nothing though, atleast I would have gained some knowledge from it. But I don't think I've been doing either of those things, and I'm p sure in the time between not playing my computer hasn't crashed or anything. I suppose it makes sense, but I'm still not sure what that means for the emulator or the data. 
Last night my friends suggested I should restart, as we tried to make a new save and that one worked just fine. I'm just worried that should I restart and try to make up the data I lost, there is still a risk of something like this happening again.

Also, I still have the .sav file but my savestates are gone, so I hope that is what you mean? ^^;
Bear with me, I'm a bit new with forums and such. But I think the guidelines said I could upload it as a png file. Just in case, I've uploaded both files to [my drive](https://drive.google.com/drive/folders/1qXqZKYxP0UJpE8LkBJzm4mVq4afF-t5h?usp=sharing).

**Comment by Dimensional** [comment]
Understood. Just remember that this is not a support forum, nor meant to be. It's an issue tracker, so if you run into a bug while using the emulator, you post it here.

**Comment by endrift** [comment]
Just looked in a hex editor; it's definitely corrupted. You might be facing drive failure.

**Comment by ghostrule** [comment]
Yup, that's my bad. I do apologize for that. I'll definitely keep it in mind next time!
About the png file, I wasn't expecting an image or anything considering it was originally a different file. But I suppose now that would of been a good indicator.
Since it seems like it wasn't an emulator problem, I'm fine if this discussion doesn't continue in this thread. You've both been really helpful, so thanks!
That's all I really needed, but I do have concerns still if either of you would like to elaborate. I'd like to go ahead and make up the data that I lost, but I'm worried about something like this happening again, especially if it was my computer/OS and not the emulator. I'm unsure if there's a way I could know or prevent it, or if I even need to worry about that happening again at all?

**Comment by Dimensional** [comment]
If the hard drive is in fact failing, then the loss of your progress is the least of your worries. A hard drive failure means even your OS will sooner or later be lost, as well as all the data on your computer.

**Comment by ghostrule** [comment]
I don't think it's that severe, considering all my other files seem to be fine. I will definitely look into it though.

**Comment by endrift** [comment]
I realized that the file you sent was a save*game* (.sav) and not a save*state* (.ss*). That being reset to all FF (as this one was) is much less weird. Do you have any of the latter floating around?

**Comment by ghostrule** [comment]
Unfortunately not, like I mentioned earlier, all of my save states were gone, and I only had the .sav file.

**Comment by zeromus** [comment]
If files which used to be on your disk, now aren't, then your system is for 100% sure melting down. Back up all your remaining files _ASAP_ and reformat your disk, at a minimum. The more you touch it before you do that, the more you break. Even better is to turn off the computer now, mount the disk on another system, and do the backup there.

**Comment by ghostrule** [comment]
Slow down guys, are you really sure this could be the case? I don't think that anything else has changed, so I don't feel like something that drastic could be happening. I will take the warning though, of course.

**Comment by ghostrule** [comment]
Just an update, I took your guys advice and poked around a bit, found what I believe to be the problem. It's since been fixed and things seem to be going smoothly now, so I'll be working on making up what I lost. I apologize again for posting in the wrong place (I'm learning!). I really appreciate the help anyway though. Thanks!

**Comment by Dimensional** [comment]
What was the cause of the problem and your fix?

**Comment by ghostrule** [comment]
I'm pretty sure it was related to a program that was installed previously... I completely forgot about it, but your guys comments convinced me to look a little deeper. I found that it was still doing stuff after I thought it had been uninstalled, and that's when I found out from others online that it was a bit malicious and shady. That's a mistake entirely on my part, but it happens. I restored my system back a bit and that cleared it up afaik.

Regarding the save file, I'm drawing conclusions that this was related to it getting corrupted/lost. I also think this because said program was trying to make changes to my computer/files. 
So in the end it wasn't related to the emulator itself at all. Again, that's my bad. Buuuut I think this case is closed?
